File: Python/underautomation/universal_robots/primary_interface/kinematics_info_package_event_args.py
import typing
from underautomation.universal_robots.primary_interface.joint_kinematics_info import JointKinematicsInfo
from underautomation.universal_robots.common.package_event_args import PackageEventArgs
import clr
import os
clr.AddReference(os.path.realpath(os.path.join(os.path.dirname(__file__), "..",  'lib', 'UnderAutomation.UniversalRobots.dll')))
from UnderAutomation.UniversalRobots.PrimaryInterface import KinematicsInfoPackageEventArgs as kinematics_info_package_event_args

class KinematicsInfoPackageEventArgs(PackageEventArgs):
	def __init__(self, _internal = 0):
		if(_internal == 0):
			self._instance = kinematics_info_package_event_args()
		else:
			self._instance = _internal
	@property
	def calibration_status(self) -> int:
		return self._instance.CalibrationStatus
	@calibration_status.setter
	def calibration_status(self, value: int):
		self._instance.CalibrationStatus = value
	@property
	def base(self) -> JointKinematicsInfo:
		return JointKinematicsInfo(self._instance.Base)
	@base.setter
	def base(self, value: JointKinematicsInfo):
		self._instance.Base = value
	@property
	def shoulder(self) -> JointKinematicsInfo:
		return JointKinematicsInfo(self._instance.Shoulder)
	@shoulder.setter
	def shoulder(self, value: JointKinematicsInfo):
		self._instance.Shoulder = value
	@property
	def elbow(self) -> JointKinematicsInfo:
		return JointKinematicsInfo(self._instance.Elbow)
	@elbow.setter
	def elbow(self, value: JointKinematicsInfo):
		self._instance.Elbow = value
	@property
	def wrist1(self) -> JointKinematicsInfo:
		return JointKinematicsInfo(self._instance.Wrist1)
	@wrist1.setter
	def wrist1(self, value: JointKinematicsInfo):
		self._instance.Wrist1 = value
	@property
	def wrist2(self) -> JointKinematicsInfo:
		return JointKinematicsInfo(self._instance.Wrist2)
	@wrist2.setter
	def wrist2(self, value: JointKinematicsInfo):
		self._instance.Wrist2 = value
	@property
	def wrist3(self) -> JointKinematicsInfo:
		return JointKinematicsInfo(self._instance.Wrist3)
	@wrist3.setter
	def wrist3(self, value: JointKinematicsInfo):
		self._instance.Wrist3 = value

File: UnderAutomation.UniversalRobots.ObfuscatedSources/PrimaryInterface/KinematicsInfoPackageEventArgs.cs
//              WARNING
// This file is an empty shell containing only public C# items.
// The internal code is hidden; to access it, you need to obtain a Source licence of the library.

using Common;

namespace PrimaryInterface {
	/// <summary>
	/// Kinematics info
	/// </summary>
	public class KinematicsInfoPackageEventArgs : PackageEventArgs, IUrDhParameters {


		public KinematicsInfoPackageEventArgs()
		{
			// Source is hidden, a Source licence is needed to access internal code...
		}

		/// <summary>
		/// Calibration status (0 : OK)
		/// </summary>
		[PackageDescription("Calibration status (0 : OK)")]
public int CalibrationStatus { get; set; }

		/// <summary>
		/// Base kinematics info
		/// </summary>
		[PackageDescription("Base kinematics info")]
public JointKinematicsInfo Base { get; set; }

		/// <summary>
		/// Shoulder kinematics info
		/// </summary>
		[PackageDescription("Shoulder kinematics info")]
public JointKinematicsInfo Shoulder { get; set; }

		/// <summary>
		/// Elbow kinematics info
		/// </summary>
		[PackageDescription("Elbow kinematics info")]
public JointKinematicsInfo Elbow { get; set; }

		/// <summary>
		/// Wrist1 kinematics info
		/// </summary>
		[PackageDescription("Wrist1 kinematics info")]
public JointKinematicsInfo Wrist1 { get; set; }

		/// <summary>
		/// Wrist2 kinematics info
		/// </summary>
		[PackageDescription("Wrist2 kinematics info")]
public JointKinematicsInfo Wrist2 { get; set; }

		/// <summary>
		/// Wrist3 (Tool) kinematics info
		/// </summary>
		[PackageDescription("Wrist3 (Tool) kinematics info")]
public JointKinematicsInfo Wrist3 { get; set; }

		/// <summary>
		/// DH parameter a2 (Shoulder.DHa)
		/// </summary>
		public double A2 { get; }

		/// <summary>
		/// DH parameter a3 (Elbow.DHa)
		/// </summary>
		public double A3 { get; }

		/// <summary>
		/// DH parameter d1 (Base.DHd)
		/// </summary>
		public double D1 { get; }

		/// <summary>
		/// DH parameter d4 (Wrist1.DHd)
		/// </summary>
		public double D4 { get; }

		/// <summary>
		/// DH parameter d5 (Wrist2.DHd)
		/// </summary>
		public double D5 { get; }

		/// <summary>
		/// DH parameter d6 (Wrist3.DHd)
		/// </summary>
		public double D6 { get; }
	}
}
